pkg/apis/core/validation/shoot_test.go

```diff
@@ -3449,6 +3449,33 @@ var _ = Describe("Shoot Validation Tests", func() {
 					},
 				}),
 			)
+
+			It("should return an error if the annotation is invalid", func() {
+				metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, "gardener.cloud/operation", "foo-bar")
+				matcher := ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeNotSupported),
+					"Field": Equal("metadata.annotations[gardener.cloud/operation]"),
+				})))
+				Expect(ValidateShoot(shoot)).To(matcher)
+			})
+		})
+
+		Context("maintenance operation validation", func() {
```

Comment:
When you move the validation in a single method, you can also move the test here to be a part of the test cases above, so that all keys that are considered valid can be tested.

Reply:
done